es/09_4_Codificando_una_P2PKH.md

```diff
@@ -0,0 +1,390 @@
+  ],
+  "vout": [
+    {
+      "value": 0.00095000,
+      "n": 0,
+      "scriptPubKey": {
+        "asm": "OP_DUP OP_HASH160 41d83eaffbf80f82dee4c152de59a38ffd0b6021 OP_EQUALVERIFY OP_CHECKSIG",
+        "hex": "76a91441d83eaffbf80f82dee4c152de59a38ffd0b602188ac",
+        "reqSigs": 1,
+        "type": "pubkeyhash",
+        "addresses": [
+          "mmX7GUoXq2wVcbnrnFJrGKsGR14fXiGbD9"
+        ]
+      }
+    },
+    {
+      "value": 0.01063793,
+      "n": 1,
+      "scriptPubKey": {
+        "asm": "OP_HASH160 b780fc2e945bea71b9ee2d8d2901f00914a25fbd OP_EQUAL",
+        "hex": "a914b780fc2e945bea71b9ee2d8d2901f00914a25fbd87",
+        "reqSigs": 1,
+        "type": "scripthash",
+        "addresses": [
+          "2N9yWARt5E3TQsX2RjsauxSZaEZVhinAS4h"
+        ]
+      }
+    }
+  ]
+}
+```
+Ahora puede mirar el `vout` `0` y ver que estaba bloqueado con la `scriptPubKey` de `OP_DUP OP_HASH160 41d83eaffbf80f82dee4c152de59a38ffd0b6021 OP_EQUALVERIFY OP_CHECKSIG`. Esa es la metodología de bloqueo estándar utilizada para una dirección P2PKH anterior con el `<pubKeyHash>` bloqueado en el medio.
+
+Ejecutarlo demostrará cómo funciona.
+
+## Ejecute un script P2PKH
+
+Cuando desbloquea un UTXO P2PKH, (efectivamente) concatena los scripts de desbloqueo y bloqueo. Para una dirección P2PKH, como el ejemplo utilizado en este capítulo, que produce:
+
+```
+Script: <signature> <pubKey> OP_DUP OP_HASH160 <pubKeyHash> OP_EQUALVERIFY OP_CHECKSIG
+```
+Con eso en conjunto, puede examinar cómo se desbloquea un UTXO P2PKH.
+
+Primero, pone las constantes iniciales en la pila, y luego hace un duplicado de la pubKey con `OP_DUP`:
+```
+Script: <signature> <pubKey> OP_DUP OP_HASH160 <pubKeyHash> OP_EQUALVERIFY OP_CHECKSIG
+Stack: [ ]
+
+Script: <pubKey> OP_DUP OP_HASH160 <pubKeyHash> OP_EQUALVERIFY OP_CHECKSIG
+Stack: [ <signature> ]
+
+Script: OP_DUP OP_HASH160 <pubKeyHash> OP_EQUALVERIFY OP_CHECKSIG
+Stack: [ <signature> <pubKey> ]
+
+Script: OP_HASH160 <pubKeyHash> OP_EQUALVERIFY OP_CHECKSIG
+Running: <pubKey> OP_DUP
+Stack: [ <signature> <pubKey> <pubKey> ]
+```
+Por qué el duplicado? Porque es necesario comprobar los dos elementos de desbloqueo: la clave pública y la firma.
+
+A continuación, `OP_HASH160` saca el resultado de la pila `<pubKey>`, lo codifica y vuelve a colocar el resultado en la pila.
+```
+Script: <pubKeyHash> OP_EQUALVERIFY OP_CHECKSIG
+Running: <pubKey> OP_HASH160
+Stack: [ <signature> <pubKey> <pubKeyHash> ]
+```
+Luego, coloca el `<pubKeyHash>` que estaba en el script de bloqueo en la pila:
+```
+Script: OP_EQUALVERIFY OP_CHECKSIG
+Stack: [ <signature> <pubKey> <pubKeyHash> <pubKeyHash> ]
+```
+`OP_EQUALVERIFY` es efectivamente dos códigos de operación: `OP_EQUAL`, que saca dos elementos de la pila y empuja `True` o `False` en función de la comparación y `OP_VERIFY` que muestra ese resultado e inmediatamente marca la transacción como inválida si es `False`. (El capítulo 12 habla más sobre el uso de `OP_VERIFY` como condicional.)
```

Comment:
``(El capítulo 12 habla más sobre el uso de `OP_VERIFY` como condicional).`` The point goes outside of the parenthesis.